### ISSUE TITLE
KoreanInput has one delay with autocomplete when using Chunjiin keyboard

### ISSUE THREAD

**Issue by sky-yoo** [flutter/flutter#40153]
Hi.

I have letter-doubling problem when I input Korean on TextField.

the problem only occur when I use Korean input system named 천지인입력기.

this is most popular input in Korea and default Samsung Galaxy Korean input.

this is the scenario
1. Type korean-elements and stop input, then Korean element combinining is auto finished about 1 second later.
2. and type another, last letter is repeated.
![problem](https://user-images.githubusercontent.com/54743309/64620062-26687c80-d41e-11e9-8c1d-29566ec51ae9.gif)

input faster and don't finish combine, then no problem.
![faster](https://user-images.githubusercontent.com/54743309/64620161-5879de80-d41e-11e9-847d-06f27267718f.gif)

input space-bar, then no problem.
![space](https://user-images.githubusercontent.com/54743309/64620207-692a5480-d41e-11e9-9579-e2c73137478e.gif)

qwert input is never finished combine, no problem.
![ezgif-3-139c64a13fa0](https://user-images.githubusercontent.com/54743309/64620219-6def0880-d41e-11e9-8a77-5bff3a96ab17.gif)

thank you.

```
flutter doctor -v
[√] Flutter (Channel stable, v1.9.1+hotfix.2, on Microsoft Windows [Version 10.0.18362.295], locale ko-KR)
    • Flutter version 1.9.1+hotfix.2 at C:\src\flutter
    • Framework revision 2d2a1ffec9 (4 days ago), 2019-09-06 18:39:49 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\Sky\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = C:\Users\Sky\AppData\Local\Android\Sdk
    • ANDROID_SDK_ROOT = C:\Users\Sky\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[!] Android Studio (version 3.4)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] VS Code, 64-bit edition (version 1.37.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.4.1

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

! Doctor found issues in 1 category.
```

**Comment by HansMuller** [comment]
This is probably related to #29341, see https://github.com/flutter/flutter/issues/29341#issuecomment-529058227, and https://github.com/flutter/flutter/issues/31512

**Comment by GaryQian** [comment]
@sky-yoo Can you confirm if this bug still exists on master? https://github.com/flutter/engine/pull/12780 addresses a suite of Samsung keyboard issues, and this may be fixed now.

**Comment by GaryQian** [comment]
I was unable to repro this both before and after the change mentioned above on a Galaxy Note 9. Do you know what device you are experiencing this on?

**Comment by sky-yoo** [comment]
@GaryQian Hello.
I checked this issue, but still exists. exactly the same.
My device is Galaxy Note 5.
and i confirmed that the problem only occured on ['천지인 키보드', '천지인 플러스 키보드', '베가 키보드'] in Samsung Korean Keyboard.


flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, v1.10.15-pre.94, on Microsoft Windows [Version 10.0.18362.418], locale ko-KR)

**Comment by GaryQian** [comment]
I have obtained additional test devices that should have this bug, I will try to repro this again!

**Comment by cwpark** [comment]
This issue occurs only on the `천지인` keyboard not the qwerty on samsung device.

And #29341 is still occurs(It's harder to repro than before) when using the korean qwerty keyboard.

steps:
1. Enter `가나`.
> 가나|
2. Move the cursor to the center.
> 가|나
3. Enter `12`.
> 가12|나
4. Move the cursor to the center.
> 가1|2나
5. Enter `가`, then:
> 가1가1가|2나

**Comment by TahaTesser** [comment]
i can reproduce this, it takes a second to autocomplete but i can't last character repeated issue 


![flutter_01](https://user-images.githubusercontent.com/48603081/86355358-317bea00-bc88-11ea-8d9a-ec00dc65c0fa.png)


<details> 
<summary>Code Sample</summary> 

``` 
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: TextField(
          decoration: InputDecoration(border: OutlineInputBorder()),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.20.0-2.0.pre, on Mac OS X 10.15.5 19F101, locale
    en-GB)
    • Flutter version 1.20.0-2.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision 15a28159bc (9 days ago), 2020-06-23 04:52:58 -0700
    • Engine revision 91a63d6a44
    • Dart version 2.9.0 (build 2.9.0-19.0.dev 7e72c9ae7e)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1

[✓] Connected device (5 available)
    • SM M305F      • 32003c30dc19668f          • android-arm64  • Android 10
      (API 29)
    • Taha’s iPhone • 00008020-001059882212002E • ios            • iOS 13.5.1
    • macOS desktop • macos                     • darwin-x64     • Mac OS X
      10.15.5 19F101
    • Web Server    • web-server                • web-javascript • Flutter Tools
    • Chrome        • chrome                    • web-javascript • Google Chrome
      83.0.4103.116

• No issues found!

``` 

</details>

**Comment by sky-yoo** [comment]
I checked this again, Issue fixed in 1.20.

```
flutter doctor -v
[√] Flutter (Channel dev, 1.20.0-7.0.pre, on Microsoft Windows [Version 10.0.18363.900], locale ko-KR)
    • Flutter version 1.20.0-7.0.pre at C:\src\flutter
    • Framework revision cc1af3afb6 (2 days ago), 2020-07-08 10:19:36 -0700
    • Engine revision 0dc86cda19
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 06cb010247)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\sky\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions.

[!] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] VS Code (version 1.46.1)
    • VS Code at C:\Users\sky\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.12.1

[!] Connected device
    ! No devices available

! Doctor found issues in 3 categories.
```

**Comment by TahaTesser** [comment]
Hi @sky-yoo
given your last message I feel safe to close this issue, if you disagree please write in the comments and I will reopen it. 
Thank you
